db/fixtures/ae_datastore/ManageIQ/Cloud/VM/Retirement/StateMachines/Methods.class/__methods__/amazon_check_pre_retirement.rb

```diff
@@ -12,7 +12,7 @@
   $evm.log('info', "Instance:<#{vm.name}> on EMS:<#{ems.try(:name)} has Power State:<#{power_state}>")
 
   # If VM is powered off or this instance is running on an instance store
-  if %w(off suspended).include?(power_state) || vm.hardware.root_device_type == "instance_store"
+  if %w(off suspended terminated).include?(power_state) || vm.hardware.root_device_type == "instance_store"
```

Comment:
@durandom 
It appears as though retirement would work okay if the instance has already been deleted from the provider except for the remove_from_provider method which would still issue the vm.remove_from_disk call.  That code should be changed to make sure there is an ems before it issues the remove call.